cmd/influxd/config.go

```diff
@@ -197,6 +197,32 @@ func (c *Config) BrokerURL() *url.URL {
 	}
 }
 
+// BrokerDir returns the data directory to start up in and does home directory expansion if necessary.
+func (c *Config) BrokerDir() string {
+	path := c.Broker.Dir
+	u, _ := user.Current()
+	hd := u.HomeDir + "/"
+
+	// if path starts wiht ~/ we need to expand it
+	if path[:2] == "~/" {
```

Comment:
Oh doh, this is just comes from a config file, so it's a string, not an OS thing.

Personally I think we should prohibit relative paths in config files. Allowing relative paths is a bomb waiting to go off IMHO.